### PR TITLE
fixed background-color for unselectable tabel  first-child cell

### PR DESCRIPTION
**Pull request by oksanashopify**
### WHY are these changes introduced?
After @shopify/polaris@12.0.0 [has been shipped](https://shopify.slack.com/archives/C4Y8N30KD/p1696832186682759) I noticed that on `main` Polaris branch IndexTable with Sub Headers does not have proper `background-color` for unselectable table  first-child cell.
image

### WHAT is this pull request doing?

I added new story "With Sub Headers Non Selectable" and fixed styles in this PR 

How it was before:
image

How it looks now:
image

[Storybook link](https://5d559397bae39100201eedc1-kabdrxeayi.chromatic.com/?path=/story/all-components-indextable--with-sub-headers-non-selectable)

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] N/A Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [x] N/A Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [x] N/A Updated the component's `README.md` with documentation changes
- [x] N/A [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris-react/src/components/IndexTable/IndexTable.scss [hunk 1]
```diff
@@ -784,14 +784,21 @@ $loading-panel-height: 53px;
...
+      position: sticky;
+    }
 
+    .TableRow-subheader {
+      .TableCell:first-child {
+        background-color: var(--p-color-bg-surface-secondary);
+      }
+    }
+  }
   .TableRow-hovered {
```
Comment (chloerice) [inline polaris-react/src/components/IndexTable/IndexTable.scss]: Noticed there's another bug where hovering disabled rows in an unselectable table gives the first cell a hover state. Could you fix that in your PR too?

```suggestion
  .TableRow-hovered:not(.TableRow-disabled) {
```

**Review by chloerice — APPROVED** [review]
Thanks @oksanashopify! 🙌🏽

**Review by mattlavio — APPROVED** [review]
🚀

File: polaris-react/src/components/IndexTable/IndexTable.scss (hunk 1 above)
Comment (oksanashopify) [reply polaris-react/src/components/IndexTable/IndexTable.scss]: Sure! Fixed)

**Comment by oksanashopify** [comment]
Hi @m4thieulavoie ! Thank you for reviewing my PR! Could you suggest me what I can do with failing CI pipelines, please, they seems to be not related to my changes. Thank you!

File: .changeset/afraid-items-compare.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': major
```
Comment (chloerice) [inline .changeset/afraid-items-compare.md]:
```suggestion
'@shopify/polaris': patch
```

Comment (chloerice) [reply .changeset/afraid-items-compare.md]: (This can target a patch release, only breaking changes need to target a major release)

Comment (oksanashopify) [reply .changeset/afraid-items-compare.md]: Got it! Thank you!

**Pull request merged**